src/plugins/tbin/tbinplugin.cpp

```diff
@@ -0,0 +1,358 @@
+#include <map>
+#include <QFile>
+#include <QFileInfo>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QTextStream>
+#include <QDebug>
+#include <sstream>
+
+namespace
+{
+    void tbinToTiledProperties( const tbin::Properties& props, Tiled::Object* obj )
+    {
+        for ( const std::pair< std::string, tbin::PropertyValue >& prop : props )
+        {
+            switch ( prop.second.type )
+            {
+                case tbin::PropertyValue::String:
+                    obj->setProperty( prop.first.c_str(), QVariant( prop.second.dataStr.c_str() ) );
+                    break;
+
+                case tbin::PropertyValue::Bool:
+                    obj->setProperty( prop.first.c_str(), QVariant( prop.second.data.b ) );
+                    break;
+
+                case tbin::PropertyValue::Float:
+                    obj->setProperty( prop.first.c_str(), QVariant( prop.second.data.f ) );
+                    break;
+
+                case tbin::PropertyValue::Integer:
+                    obj->setProperty( prop.first.c_str(), QVariant( prop.second.data.i ) );
+                    break;
+            }
+        }
+    }
+
+    void tiledToTbinProperties( const Tiled::Object* obj, tbin::Properties& props )
+    {
+        for ( auto it = obj->properties().cbegin(); it != obj->properties().cend(); ++it )
+        {
+            tbin::PropertyValue prop;
+            if ( it.value().type() == QVariant::Bool )
+            {
+                prop.type = tbin::PropertyValue::Bool;
+                prop.data.b = it.value().toBool();
+            }
+            else if ( it.value().type() == QMetaType::Float )
+            {
+                prop.type = tbin::PropertyValue::Float;
+                prop.data.f = it.value().toFloat();
+            }
+            else if ( it.value().type() == QVariant::Int )
+            {
+                prop.type = tbin::PropertyValue::Integer;
+                prop.data.i = it.value().toInt();
+            }
+            else if ( it.value().type() == QVariant::String )
+            {
+                prop.type = tbin::PropertyValue::String;
+                prop.dataStr = it.value().toString().toStdString();
+            }
+            else throw std::invalid_argument( "Unsupported property type" );
+            props.insert( std::make_pair( it.key().toStdString(), prop ) );
+        }
+    }
+}
+
+namespace Tbin {
+
+void TbinPlugin::initialize()
+{
+    addObject(new TbinMapFormat(this));
+}
+
+
+TbinMapFormat::TbinMapFormat(QObject *parent)
+{
+}
+
+Tiled::Map *TbinMapFormat::read(const QString &fileName)
+{
+    QFile file(fileName);
+    if (!file.open(QIODevice::ReadOnly)) {
+        mError = tr("Could not open file for reading.");
+        return nullptr;
+    }
+
+    QByteArray contents = file.readAll();
+    std::istringstream ss( std::string( contents.constData(), contents.length() ) );
+
+    tbin::Map tmap;
+    Tiled::Map* map;
+    try
+    {
+        tmap.loadFromStream( ss );
```

Comment:
Coding style: please don't put spaces on the inner sides of parentheses (you're fine to keep them in the tbin library part, of course).